cloudinit/config/cc_resizefs.py

```diff
@@ -146,6 +148,42 @@ def _can_skip_resize_ufs(mount_point, devpth):
 RESIZE_FS_PRECHECK_CMDS = {"ufs": _can_skip_resize_ufs}
 
 
+def get_device_info_from_zpool(zpool) -> Optional[str]:
+    # zpool has 10 second timeout waiting for /dev/zfs LP: #1760173
+    container = util.is_container()
+    if not os.path.exists("/dev/zfs"):
+        LOG.debug("Cannot get zpool info, no /dev/zfs")
+        return None
+    try:
+        zpoolstatus, err = subp.subp(["zpool", "status", zpool])
+        if err:
+            LOG.info(
+                "zpool status returned error: [%s] for zpool [%s]",
+                err,
+                zpool,
+            )
+            return None
+    except subp.ProcessExecutionError as err:
+        if container:
+            LOG.info(
+                "Unable to get zpool status of %s: %s in container", zpool, err
+            )
+            return None
+        LOG.warning("Unable to get zpool status of %s: %s", zpool, err)
+        return None
+    r = r".*(ONLINE).*"
+    for line in zpoolstatus.split("\n"):
+        if re.search(r, line) and zpool not in line and "state" not in line:
+            disk = line.split()[0]
+            LOG.debug('found zpool "%s" on disk %s', zpool, disk)
+            return disk
+    if container:
+        LOG.info("No zpool found: %s: %s", zpool, err)
```

Comment:
In this case, we probably want to log zpoolstatus as well because we were unable to parse it.